crates/libtiny_tui/src/input_area/mod.rs

```diff
@@ -333,110 +319,122 @@ impl InputArea {
                 }
                 WidgetRet::KeyHandled
             }
-
-            ////////////////////////////////////////////////////////////////////
-            // Scrolling in history or autocompletion list
-            Key::Arrow(Arrow::Up) => {
-                // invalidate height calculation
-                self.height = None;
-                let mode = mem::replace(&mut self.mode, Mode::Edit);
-
-                match mode {
-                    Mode::Edit => {
-                        if !self.history.is_empty() {
-                            self.mode = Mode::History((self.history.len() as i32) - 1);
-                            self.move_cursor_to_end();
-                        }
-                    }
-                    Mode::History(hist_curs) => {
-                        self.mode = Mode::History(if hist_curs > 0 {
-                            hist_curs - 1
-                        } else {
-                            hist_curs
-                        });
-                        self.move_cursor_to_end();
-                    }
-                    Mode::Autocomplete {
-                        original_buffer,
-                        insertion_point,
-                        word_starts,
-                        completions,
-                        current_completion,
-                        ..
-                    } => {
-                        let current_completion = if current_completion == completions.len() - 1 {
-                            0
-                        } else {
-                            current_completion + 1
-                        };
-
-                        let cursor =
-                            (insertion_point + completions[current_completion].len()) as i32;
-
-                        self.mode = Mode::Autocomplete {
-                            original_buffer,
-                            insertion_point,
-                            word_starts,
-                            completions,
-                            current_completion,
-                        };
-
-                        self.move_cursor(cursor);
+            _ => {
+                match key {
+                    Key::Char(ch) => {
+                        self.modify();
+                        self.buffer.insert(self.cursor as usize, ch);
+                        self.inc_cursor();
+                        WidgetRet::KeyHandled
                     }
-                }
+                    ////////////////////////////////////////////////////////////////////
+                    // Scrolling in history or autocompletion list
+                    Key::Arrow(Arrow::Up) => {
```

Comment:
Why was this not mapped to a `KeyAction`?

Reply:
Yes, this should be added too. What should the `KeyAction` be called since it covers different modes?

Reply:
I think we could treat these modes (next/prev in history or next/prev in autocompletion) as the same. For naming, maybe `NextCompletion` (up) and `PrevCompletion` (down)? Or `NextEntry`/`PrevEntry`?

Reply:
Ah, I see now why I didn't add it. Line 721, `fn autocomplete()` in `input_area/mod.rs` there is a hardcoding of 
```
self.keypressed(Key::Arrow(Arrow::Up), Some(KeyAction::InputAutoComplete));
```

Reply:
I'm a bit confused.. I think `InputAutoComplete` is not used anywhere (we don't check for it). Couldn't we remove it?

Reply:
We do check for it:
```
            Some(KeyAction::InputAutoComplete) => {
                if self.exit_dialogue.is_none() {
                    self.input_field.autocomplete(&self.nicks);
                }
                WidgetRet::KeyHandled
            }
```
I will add the `NextEntry/PrevEntry` key actions

Reply:
After adding `NextEntry`/`PrevEntry` could we also add one for the characters, and remove the `key` argument from `InputArea::keypressed`? It seems like we won't need it anymore if we also have a case for single characters.

Reply:
Was thinking the same. A `KeyAction::Input(char)` or something?

Reply:
That works. We can rename it later if needed.